### PR TITLE
Remove use of subscribe as site subscriber

### PR DESCRIPTION
**Pull request by millerf**
Fixes https://github.com/Automattic/gold/issues/190

## Proposed changes:

* Do not rely on subscribe_as_site_subscriber for tiers

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* See WPCOM PR
* Unit tests
```
➜ cd projects/plugins/jetpack                                                     
➜  jetpack git:(add/remove-user-of-subscribe_as_site_subscriber) pnpm test-extensions extensions/store/membership-products/test/selectors-test.js
```

```
 ➜ jetpack docker phpunit-multisite --php 7.0  --  --testsuite=subscriptions 
``

### CODE REVIEW

File: projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php [hunk 1]
```diff
@@ -30,6 +30,20 @@ class Jetpack_Memberships {
...
+	 * @var string
+	 */
+	public static $tier_type = 'tier';
+
+	/**
+	 * Tier type for plans
+	 *
+	 * @var string
+	 */
+	public static $tier_type = 'tier';
```
Comment (chrismccluskey) [inline projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php]: This seems to have been added by mistake.

**Review by chrismccluskey — CHANGES_REQUESTED** [review]
Looks good aside from the duplicated variable.

File: projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php (hunk 1 above)
Comment (millerf) [reply projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php]: Nice cath! Fixed

**Review by chrismccluskey — DISMISSED** [review]
LGTM

**Review by chrismccluskey — DISMISSED** [review]

**Review by chrismccluskey — DISMISSED** [review]

**Review by chrismccluskey — APPROVED** [review]
LGTM

**Pull request merged**